File: backend/src/main/java/com/yourwatchrental/watchrental/rental/PaymentStatus.java
package com.yourwatchrental.watchrental.rental;

public enum PaymentStatus {
    PENDING,     // oczekuje na płatność
    SUCCESSFUL,  // płatność zakończona poprawnie
    FAILED,      // płatność nieudana
    ON_SPOT,
    CANCELLED,
    REFUNDED
}

File: frontend/src/app/core/models/rentals/payment-status.ts
export enum PaymentStatus{
    PENDING = "PENDING",
    SUCCESSFUL = "SUCCESSFUL",
    FAILED = "FAILED",
    ON_SPOT = "ON_SPOT",
    CANCELLED = "CANCELLED",
    REFUNDED = "REFUNDED"
}

export const PaymentStatusLabel: Record<PaymentStatus, string> = {
    [PaymentStatus.PENDING]: "PENDING",
    [PaymentStatus.SUCCESSFUL]: "SUCCESSFUL",
    [PaymentStatus.FAILED]: "FAILED",
    [PaymentStatus.ON_SPOT]: "ON SPOT",
    [PaymentStatus.CANCELLED]: "CANCELLED",
    [PaymentStatus.REFUNDED]: "REFUNDED"
};